### PR TITLE
Deprecate /start/add-domain

### PR DESCRIPTION
**Pull request by southp**
## Proposed Changes

~This PR deprecates `/start/add-domain` flow and `/start/site-selected` flows since they are no longer in use.~ (`/start/site-selected` is not deprecated here. See the discussion below)

This PR deprecates `start/add-domain` since it's no longer in use.

## Testing Instructions

What's needed here is to confirm these two flows are not in use. There are two code references for the site-selected flow in wpcom, but they both don't look in use. By inspecting the Tracks Trend view of `calyspo_signup_start` 2fe39-pb, there is no production event with the flow prop as `add-domain` or `selected-site`. 

Please feel free to verify in some other ways; I might be overlooking something :)  

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by aneeshd16 — CHANGES_REQUESTED** [review]
I see that the `site-selected` flow is used here:https://github.com/Automattic/wp-calypso/blob/4d14c0099897cb500f87c026bb09ba71226b08c7/client/my-sites/domains/paths.js#L354-L358
and this function is used in a few different places:
* https://github.com/Automattic/wp-calypso/blob/7e80a577dcc720d60ccd5051ebd839a7afdcd482/client/my-sites/checkout/checkout-thank-you/domains/thank-you-content/domain-registration.tsx#L52
* https://github.com/Automattic/wp-calypso/blob/d63721736784ff9803ba3cf3c105cf7a1ff93da1/client/my-sites/domains/domain-management/list/domain-only-upsell-carousel.tsx#L219
* etc..

I'm also seeing Tracks events for this flow(2fe69-pb/#plain) We might need to hold off on removing this flow for now.

The `add-domain` flow seems okay for removal, I'm not seeing any Tracks events for that. I found a few more references that we can remove:
add-domain:
* https://github.com/Automattic/wp-calypso/blob/24c96cd1559e664ed0e937f42648d172ad942f07/client/signup/main.jsx#L360
* https://github.com/Automattic/wp-calypso/blob/0144ea16c1b0f255acf1a530c651403effd73e24/client/signup/config/flows.js#L24
* https://github.com/Automattic/wp-calypso/blob/4fd2d5cb8ce768b9245328753e2e6324623ce582/client/signup/config/steps-pure.js#L425

**Comment by southp** [comment]
Nice finding, @aneeshd16 . I don't know why I didn't find those events when I searched. The flow looks very broken now but I think we need to investigate further before deprecating it:
image

I'll also work on removing a few more references of `add-domain` flow.

**Comment by southp** [comment]
@aneeshd16 I've updated the code and the title/description of this PR. It's now ready for re-review :)

File: client/signup/main.jsx [hunk 1]
```diff
@@ -357,7 +357,7 @@ class Signup extends Component {
 		const flowName = this.props.flowName;
 		// p2v1 also has a user step at the end but the flow is otherwise broken.
 		// reader also has a user step at the end, but this change doesn't fix that flow.
-		const eligbleFlows = [ 'domain', 'add-domain' ];
+		const eligbleFlows = [ 'domain' ];
 		if ( ! eligbleFlows.includes( flowName ) || ! this.props.progress ) {
```
Comment (aneeshd16) [inline client/signup/main.jsx:361]: Minor typo :) 
```suggestion
		const eligibleFlows = [ 'domain' ];
		if ( ! eligibleFlows.includes( flowName ) || ! this.props.progress ) {
```

**Review by aneeshd16 — APPROVED** [review]
Thanks for the fixes, LGTM!

File: client/signup/main.jsx (hunk 1 above)
Comment (southp) [reply client/signup/main.jsx:361]: My English lol.

**Pull request merged**